src/javascript/app_2/pages/trading/components/elements/popups.jsx

```diff
@@ -0,0 +1,117 @@
+                            className={this.props.className}
+                            href='javascript:;'
+                            onClick={this.toggleModal}
+                        >
+                            {this.props.text}
+                        </a>
+                    }
+                </React.Fragment>
+                }
+                <Modal
+                    show={this.state.isOpen}
+                    onClose={this.toggleModal}
+                    action_one={this.props.action_one}
+                    action_one_text={this.props.action_one_text}
+                    action_two={this.props.action_two}
+                    action_two_text={this.props.action_two_text}
+                >
+                    {this.props.children}
+                </Modal>
+            </React.Fragment>
+        );
+    }
+}
+
+class Modal extends React.PureComponent {
+    render() {
+        const visibility = {
+            visibility: this.props.show ? 'visible' : 'hidden',
+        };
+        const popup_backdrop_class = classNames('popup-backdrop', {
+            show: this.props.show,
+        });
+        const popup_class = classNames('popup', {
+            show: this.props.show,
+        },  this.props.className );
+        return (
+            <div
+                className={popup_backdrop_class}
+                style={visibility}
+            >
+                <div
+                    className={popup_class}
+                    style={visibility}
+                >
+                    <div className='popup-close' onClick={this.props.onClose}>
+                        <img src={Url.urlForStatic('images/trading_app/close.svg')} alt='Close' />
+                    </div>
+                    {!!this.props.title &&
+                        <div className='popup-title'>{this.props.title}</div>
+                    }
+                    <div className='popup-content'>
+                        {this.props.children}
+                    </div>
+                    <div className='popup-footer'>
+                        <Button
+                            className='flat'
+                            text={this.props.action_one_text}
+                            onClick={this.props.action_one}
+                            has_effect
+                        />
+                        <Button
+                            className='flat'
+                            text={this.props.action_two_text}
+                            onClick={this.props.action_two}
+                            has_effect
+                        />
+                    </div>
+                </div>
+            </div>
+        );
+    }
+}
+
+// TODO: Remove defaultProps dummy values
+Modal.defaultProps = {
+    title: 'Title Goes Here',
+};
+
+PopUp.defaultProps = {
+    is_btn         : false,
+    text           : 'Modal Button',
+    autostart      : false,
+    action_one_text: 'Action 1',
+    action_two_text: 'Action 2',
```

Comment:
for default text, maybe we use 'Cancel' and 'OK' instead of 'Action 1' and 'Action 2' 🤔

Reply:
I asked the UI/UX the same thing, but probably best to wait if in all cases (once Static Pages are finalized) we're going to use 'OK' and 'Cancel'. 

We're going to remove `defaultProps` either way. :)

Reply:
ahh, I see.. sure 👍